File: Training-Management/server/training_spring/src/main/java/com/netlink/server/model/user/Userdata.java
package com.netlink.server.model.user;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Userdata {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    private String nsid;
    private String username;
    private String password;
    private String fname;
    private String lname;
    private String email;
    private Long phone;

    @ManyToOne
    @JoinColumn(name = "department_id",referencedColumnName = "id")
    @JsonIgnoreProperties("userdataList")
    private Department department;


    @ManyToOne
    @JoinColumn(name = "designation_id",referencedColumnName = "id")
    @JsonIgnoreProperties("userdataList")
    private Designation designation;



    @ManyToOne
    @JoinColumn(name = "role_id",referencedColumnName = "id")
    @JsonIgnoreProperties("userdataList")
    private Role role;

    private Boolean active;
    private String profile;
    private Integer otp;
    private Boolean confirmotp;

}

File: Training-Management/client/TrainingManagement/src/app/model/UserData.ts
import { Department } from "./Department";
import { Designation } from "./Designation";
import { Role } from "./Role";



export class User{
   
    userId:string;
    nsid:string;
    password:string;
    fname:string;
    lname:string;
    email:string;
    phone:number;
    department:Department;
    designation:Designation;
    role:Role;
    active:Boolean;
    profile:string;
    otp:number;
    confirmotp:number;
}